node/flatpak-node-generator.py

```diff
@@ -1171,6 +1229,10 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         yield from self.process_dependencies(lockfile, data.get('dependencies', {}))
 
 
+class NpmRCFileProvider(RCFileProvider):
+    RCFILE_NAME = '.npmrc'
```

Comment:
Having the two separate classes that only change a single constant on them feels a bit...off. It might be a bit cleaner if:

- There's one `RcFileProvider` that takes the file name as an instance variable.
  - Add two instance methods, `create_npm` and `create_yarn`, that do what it sounds like they do.
- Then `create_rcfile_providers` can just call those two methods, but only one actual class is needed.

Reply:
This is mostly for future. It's not unlikely that I've missed some npm/yarn-specific quirks that we'll need to handle.